Tasks/Common/VstsAzureRestHelpers_/VstsAzureRestHelpers_.psm1

```diff
@@ -661,24 +674,54 @@ function Remove-AzureSqlDatabaseServerFirewallRule
           [String] [Parameter(Mandatory = $true)] $firewallRuleName)
 
     Trace-VstsEnteringInvocation $MyInvocation
-    Write-Verbose "Removing firewall rule $firewallRuleName on azure database server: $serverName"
-
-    $connectionType = Get-ConnectionType -serviceEndpoint $endpoint
 
-    if(IsLegacyAzureConnection $connectionType)
-    {
-        Remove-LegacyAzureSqlServerFirewall -endpoint $endpoint -serverName $serverName -firewallRuleName $firewallRuleName
-    }
-    elseif (IsAzureRmConnection $connectionType)
+    try
     {
-        Remove-AzureRmSqlServerFirewall -endpoint $endpoint -serverName $serverName -firewallRuleName $firewallRuleName
+        Write-Verbose "Removing firewall rule $firewallRuleName on azure database server: $serverName"
+
+        $connectionType = Get-ConnectionType -serviceEndpoint $endpoint
+
+        if(IsLegacyAzureConnection $connectionType)
+        {
+            Remove-LegacyAzureSqlServerFirewall -endpoint $endpoint -serverName $serverName -firewallRuleName $firewallRuleName
+        }
+        elseif (IsAzureRmConnection $connectionType)
+        {
+            Remove-AzureRmSqlServerFirewall -endpoint $endpoint -serverName $serverName -firewallRuleName $firewallRuleName
+        }
+        else
+        {
+            throw (Get-VstsLocString -Key AZ_UnsupportedAuthScheme0 -ArgumentList $connectionType)
+        }
+
+        Write-Verbose "Removed firewall rule $firewallRuleName on azure database server: $serverName"
     }
-    else
+    catch 
     {
-        throw (Get-VstsLocString -Key AZ_UnsupportedAuthScheme0 -ArgumentList $connectionType)
+        $parsedException = Parse-Exception($_.Exception)
+        if($parsedException)
+        {
+            throw  $parsedException
+        }
+        throw $_.Exception.ToString()
     }
+}
 
-    Write-Verbose "Removed firewall rule $firewallRuleName on azure database server: $serverName"
+function Parse-Exception($exception){
+    if($exception) {
+        $response = $exception.Response
+        if($response) {
+            $responseStream =  $response.GetResponseStream()
```

Comment:
Have we left any field here , apart fro response?
If yes, we should add that as well to capture more info as its an exception & more info will only help.